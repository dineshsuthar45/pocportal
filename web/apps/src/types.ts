import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
