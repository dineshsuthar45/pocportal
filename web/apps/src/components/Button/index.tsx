import { Button } from "@mui/material";
import { IGlobalButtonProps } from "../../types";

const ContainedButton = ({
  label,
  onClick,
  disabled,
  style,
  type,
}: IGlobalButtonProps) => {
  const buttonType = type === "reset" || type === "submit" ? type : "button";
  return (
    <Button
      variant="contained"
      color="primary"
      type={buttonType}
      disabled={disabled}
      onClick={onClick}
      sx={{ width: "fit-content" }}
      style={{ ...style }}
    >
      {label}
    </Button>
  );
};

const OutlinedButton = ({
  label,
  onClick,
  disabled,
  style,
  type,
}: IGlobalButtonProps) => {
  const buttonType = type === "reset" || type === "submit" ? type : "button";
  return (
    <Button
      variant="outlined"
      color="primary"
      type={buttonType}
      disabled={disabled}
      onClick={onClick}
      sx={{ width: "fit-content" }}
      style={{ ...style }}
    >
      {label}
    </Button>
  );
};

const TextButton = ({
  label,
  onClick,
  disabled,
  style,
}: IGlobalButtonProps) => {
  return (
    <Button
      variant="text"
      color="primary"
      onClick={onClick}
      disabled={disabled}
      sx={{
        width: "fit-content",
        color: "#405270",
        "&:hover": {
          color: "green",
          textDecoration: "underline",
        },
      }}
      style={{ ...style }}
    >
      {label}
    </Button>
  );
};

export { ContainedButton, OutlinedButton, TextButton };
