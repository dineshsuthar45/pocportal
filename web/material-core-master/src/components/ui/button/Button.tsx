import MuiButton from "@mui/material/Button";
import { ButtonProps } from "./Button.types";
import classNames from "classnames";
import { useTheme } from "@mui/material";

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/api/button/
 */
export const MButton = ({
  label,
  color = "secondary",
  disabled = false,
  disableElevation = false,
  disableFocusRipple = false,
  disableRipple = false,
  fullWidth = false,
  href = "",
  size = "medium",
  variant = "outlined",
  className,
  ...props
}: ButtonProps) => {
  const theme = useTheme();

  return (
    <MuiButton
      className={classNames("button-root", theme.palette.mode, className)}
      type="button"
      color={color}
      disabled={disabled}
      disableElevation={disableElevation}
      disableFocusRipple={disableFocusRipple}
      disableRipple={disableRipple}
      fullWidth={fullWidth}
      href={href}
      size={size}
      variant={variant}
      {...props}
    >
      {label}
    </MuiButton>
  );
};

MButton.defaultProps = {
  color: "secondary",
  disabled: false,
  disableElevation: false,
  disableFocusRipple: false,
  disableRipple: false,
  fullWidth: false,
  href: "",
  size: "medium",
  variant: "outlined",
};
