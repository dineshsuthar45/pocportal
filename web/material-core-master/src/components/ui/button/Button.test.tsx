import { render } from "@testing-library/react";
import React from "react";
import { MButton } from "./Button";
import { ButtonProps } from "./Button.types";

describe("Button component", () => {
  const renderComponent = () => render(<MButton {...props} />);
  let props: ButtonProps;

  beforeEach(() => {
    props = {
      label: "Place Order",
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
