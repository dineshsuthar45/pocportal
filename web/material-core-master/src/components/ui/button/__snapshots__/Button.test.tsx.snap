// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium button-root light css-1l9sosn-MuiButtonBase-root-MuiButton-root"
    href=""
    tabindex="0"
    type="button"
  >
    Place Order
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
